src/App.tsx

```diff
@@ -1,26 +1,86 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { deleteTodo, getTodos, USER_ID } from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { ErrorField } from './components/ErrorField/ErrorField';
+import { FilterState } from './types/FilterState';
+import { getPreparedTodos } from './utils/getPreperedTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [activeFilter, setActiveFilter] = useState<FilterState>(
+    FilterState.ALL,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState([0]);
+
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+  const completedTodosCount = todos.filter(todo => todo.completed).length;
+  const preparedTodos = getPreparedTodos(todos, activeFilter);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoadingTodos(prev => [...prev, todoId]);
+    deleteTodo(todoId)
+      .then(() =>
+        setTodos((prev: Todo[]) => prev.filter(todo => todo.id !== todoId)),
+      )
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
```

Comment:
will be great to create enum with error messages, try not to use magic string in code